samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -92,21 +92,55 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
   }
 
   /**
-   * Getter for event time
+   * Getter for event time (Nanos)
    * @return this.eventTime
    */
-  public long getEventTime() {
+  public long getEventTimeNanos() {
```

Comment:
Have you seen scenarios where event time is in nanos? It's not common to have nanosecond resolution clocks.

Reply:
we need to use higher than Millis resolution to compute query latency in our case. 
One idea is to use Millis for eventTime, and keep its semantics, but use nanoArrivalTime so that we can calculate accurate queryLatency.

Reply:
will restore EpochMillis for eventTime, but will have arrivalSystemNanoTime to replace arrivalTime and it will be nanoTime

Reply:
done